controllers/main.js

```diff
@@ -53,21 +68,21 @@ module.exports.sendPlayData = async (req, res, next) => {
 
 module.exports.updatePlayData = async (req, res, next) => {
   const sessionId = req.body.sessionId;
-  const nextPageNum = parseInt(req.body.pageNum) + 1;
-  const guessData = {
-    gameName: req.body.gameName,
-    imgLink: req.body.imageLink,
-    userGuess: req.body.guess === req.body.gameName,
-  };
+  const pageNum = parseInt(req.body.pageNum);
+  const userGuess = req.body.guess === req.body.gameName;
+  const elemId = req.body.elemId;
   await PlaySession.updateOne(
     { sessionId: sessionId },
-    { $push: { sessionData: guessData } }
+    { $set: { "sessionData.$[elem].userGuess": userGuess } },
+    { arrayFilters: [{ "elem._id": elemId }] }
```

Comment:
I was kinda grabbing at straws here trying to figure out how to access the specific key that i want to assign value to. The only solution that ended up working in the end was passing the specific play page's id from mongodb and the using it to assign the value the way i wanted.
It feels like one of those "tiesiai per aplinkui" approaches, perhaps there is a better way?

Reply:
I'm not too sure how to solve this issue in a better way because unfortunately I don't remember much about Mongo, but for now you can leave it like this